src/pure_rworset.erl

```diff
@@ -0,0 +1,270 @@
+%%
+%% Copyright (c) 2015-2016 Christopher Meiklejohn.  All Rights Reserved.
+%% Copyright (c) 2007-2012 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+%% @doc Pure RWORSet CRDT: pure op-based rmv-wins observed-remove set
+%%
+%% @reference Carlos Baquero, Paulo Sérgio Almeida, and Ali Shoker
+%%      Making Operation-based CRDTs Operation-based (2014)
+%%      [http://haslab.uminho.pt/ashoker/files/opbaseddais14.pdf]
+
+-module(pure_rworset).
+-author("Georges Younes <georges.r.younes@gmail.com>").
+
+-behaviour(type).
+-behaviour(pure_polog).
+
+-define(TYPE, ?MODULE).
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+-export([new/0, new/1]).
+-export([mutate/3, query/1, equal/2]).
+-export([redundant/2, remove_redundant_crystal/2, remove_redundant_polog/2, check_stability/2]).
+
+-export_type([pure_rworset/0, pure_rworset_op/0]).
+
+-opaque pure_rworset() :: {?TYPE, payload()}.
+-type payload() :: {pure_type:polog(), ordsets:set()}.
+-type pure_rworset_op() :: {add, pure_type:element()} | {rmv, pure_type:element()}.
+
+%% @doc Create a new, empty `pure_rworset()'
+-spec new() -> pure_rworset().
+new() ->
+    {?TYPE, {orddict:new(), ordsets:new()}}.
+
+%% @doc Create a new, empty `pure_rworset()'
+-spec new([term()]) -> pure_rworset().
+new([]) ->
+    new().
+
+-spec redundant({pure_type:id(), pure_rworset_op()}, {pure_type:id(), pure_rworset_op()}) ->
```

Comment:
Now there are macros with two letters. The first letter means Remove or Add/Keep for the first element. The second for the second.